lib/srv/desktop/discovery.go

```diff
@@ -122,7 +122,15 @@ func (s *WindowsService) getDesktopsFromLDAP() types.ResourcesWithLabels {
 	s.cfg.Log.Debugf("searching for desktops with LDAP filter %v", filter)
 
 	entries, err := s.lc.readWithFilter(s.cfg.DiscoveryBaseDN, filter, computerAttribtes)
-	if err != nil {
+	if trace.IsConnectionProblem(err) {
+		// If the connection was broken, re-initialize the LDAP client so that it's
+		// ready for the next reconcile loop. Return the last known set of desktops
+		// in this case, so that the reconciler doesn't delete the desktops it already
+		// knows about.
+		s.cfg.Log.Info("LDAP connection error when searching for desktops, reinitializing client")
+		s.initializeLDAP()
```

Comment:
Looks like this error isn't being handled. Logging it might be the only actionable thing to do here though.

Reply:
Yeah, I was thinking I would avoid spamming with the same error message in case this takes several retries before it succeeds, but I'm not opposed to logging it.